common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -311,6 +318,37 @@ static void copyMemory(Object src, long srcOffset, Object dst, long dstOffset, l
         }
     }
 
+    static boolean equals(final byte[] bytes1, final byte[] bytes2) {
+        if (bytes1.length != bytes2.length) {
+            return false;
+        }
+        if (bytes1.length == 0) {
+            return true;
+        }
+        int remainingBytes = bytes1.length & 7;
+        for (int i = bytes1.length - 8; i >= remainingBytes; i -= 8) {
+            final long word1 = UNSAFE.getLong(bytes1, ARRAY_BASE_OFFSET + i);
```

Comment:
> > to communicate conceptual meaning of values with local variable then can we use a comment

> I disagree sooo much :-) Don't use comments when you can just name your variable the right way...

Agreed that variable naming is important, but variables are not always necessary.

> is this after javac?

Yes. Here is the output from hotspot with `-XX:+PrintCompilation` enabled. `equalsWithTemps` (which is what was originally here) and `equalsInlined` (which is with inlined suggestions).

```java
    public static final void main(String[] args) throws InterruptedException {
        final int len = 100;
        byte[] foo = new byte[len];
        byte[] goo = new byte[len];
        for (int i = 0; i < foo.length; ++i) {
            foo[i] = goo[i] = (byte) i;
        }

        doTestLoop(foo, goo);
        Thread.sleep(2000);
        doTestLoop(foo, goo);
    }

    static void doTestLoop(byte[] foo, byte[] goo) {
        int x = 0;
        for (int i = 0; i < 1000000; ++i) {
            if (equalsWithTemps(foo, goo)) {
                ++x;
            }
            if (equalsInlined(foo, goo)) {
                ++x;
            }
        }
        System.err.println("x: " + x);
    }
```

```
    132   55       3       com.test::equalsWithTemps (189 bytes)
    132   56       3       com.test::equalsInlined (166 bytes)
    132   57       4       com.test::equalsWithTemps (189 bytes)
    132   58       4       com.test::equalsInlined (166 bytes)
    135   56       3       com.test::equalsInlined (166 bytes)   made not entrant
    135   55       3       com.test::equalsWithTemps (189 bytes)   made not entrant
    137   59 %     3       com.test::doTestLoop @ 7 (61 bytes)
    138   60       3       com.test::doTestLoop (61 bytes)
    138   61 %     4       com.test::doTestLoop @ 7 (61 bytes)
    147   59 %     3       com.test::doTestLoop @ -2 (61 bytes)   made not entrant
    157   61 %     4       com.test::doTestLoop @ -2 (61 bytes)   made not entrant
x: 2000000
   2163   62 %     4       com.test::doTestLoop @ 7 (61 bytes)
   2172   63       4       com.test::doTestLoop (61 bytes)
   2181   60       3       com.test::doTestLoop (61 bytes)   made not entrant
x: 2000000
```

Although it appears the JIT is compiling these ...not sure how to get more visibility than this (besides going to [PrintAssembly](https://wikis.oracle.com/display/HotSpotInternals/PrintAssembly)).  All I can see is the number of bytes for the byte code is higher when the variables are not inlined.

> this is too small of a change to invest much time discussing :-)

Agreed, but good to understand how other layers that we rely on behave.  BTW do you know of any other useful flags?  I tried `-XX:+UnlockDiagnosticVMOptions -XX:+PrintInlining` but that didn't really provide anything useful here...unsafe is already intrinsic.